LXStudio-IDE/src/main/java/titanicsend/pattern/tom/Fire.java

```diff
@@ -0,0 +1,95 @@
+package titanicsend.pattern.tom;
+
+import heronarts.lx.LX;
+import heronarts.lx.color.LXColor;
+import heronarts.lx.model.LXPoint;
+import heronarts.lx.parameter.BooleanParameter;
+import heronarts.lx.parameter.BoundedParameter;
+import heronarts.lx.parameter.CompoundParameter;
+import heronarts.lx.parameter.LXParameter;
+import titanicsend.pattern.PeriodicPattern;
+
+import static processing.core.PApplet.lerpColor;
+
+public class Fire extends PeriodicPattern {
+    private float ROW_HEIGHT = 100000;
+    private float COLUMN_WIDTH = 100000;
+    private int NUM_COLUMNS;
+    private int NUM_ROWS;
+    private int[][] buffer;
+    private int[] gradient;
+
+    protected final CompoundParameter fuel = (CompoundParameter)
+            new CompoundParameter("Fuel", 1);
+
+    public Fire(LX lx) {
+        super(lx);
+        NUM_ROWS = calculateRows();
+        NUM_COLUMNS = calculateColumns();
+        gradient = calculateGradient(LXColor.rgb(0xcf, 0x6f, 0x0f), 36);
```

Comment:
TODO: refactor `TEPattern` so I can take advantage of both PeriodicPattern and TEColorGroupPattern here so we can change the color in the UI.